### DISCUSSION TITLE
Cannot find module '@napi-rs/snappy-linux-x64-gnu'

### DISCUSSION THREAD

**Discussion by sambitraze** [directus/directus#16257] [Archive]
I was running bootstrap command and i got this
 npm run pre

> vedanidhi@1.0.0 pre
> npx directus bootstrap

Error: Cannot find module '@napi-rs/snappy-linux-x64-gnu'
Require stack:

**Comment by rijkvanzanten** [comment]
How did you install Directus? Can you try installing the `directus` package again fresh? `npm install directus`.

**Reply by sambitraze** [reply]
see I developed the entire project on my windows system then I am migrating it to ubuntu 20.04 and 64 x 86 ec2 instance.
so just like another of my previous directus project I created the DB and bd users then I ran npm Install then npx directus bootstrap

**Reply by sambitraze** [reply]
```json
{
  "name": "vedanidhi",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "start": "directus start",
    "pre": "npx directus bootstrap",
    "restore": "npx directus schema apply snapshot.yaml",
    "backup": "npx directus schema snapshot snapshot.yaml"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "dependencies": {
    "directus": "^9.19.2",
    "firebase-admin": "^11.2.0",
    "pg": "^8.8.0"
  }
}
```

**Reply by sambitraze** [reply]
my package json

**Comment by Voldemorten** [comment]
I'm having same issue. I can run it just fine locally, but when I deploy it on Digital Ocean I get the same error.

**Reply by Voldemorten** [reply]
This is the full error stack:
[nofipa-directus-deploy.txt](https://github.com/directus/directus/files/10000483/nofipa-directus-deploy.txt)

**Comment by Voldemorten** [comment]
Okay, so I finally solved this if anyone is interested. I added the snappy package as a custom dependency instead of relying on directus' dependencies. I don't know why this works, because the snappy library was added in node_modules through directus, but this works.

**Reply by Voldemorten** [reply]
@sambitraze

**Comment by thomasjonas** [comment]
I have the same problem. When I run `npm init directus-project example-project` I get `Error: Cannot find module '@napi-rs/snappy-darwin-arm64'`

**Reply by Voldemorten** [reply]
Have you tried adding it as it's own dependency as suggested above?

**Reply by thomasjonas** [reply]
Yeah, that fixed it! So in the end I ran:
`npm init directus-project example-project`

then got the error: `Error: Cannot find module '@napi-rs/snappy-darwin-arm64'`
from the `example-project` directory I ran `npm i @napi-rs/snappy-darwin-arm64` followed by `npx directus init`.

**Reply by Voldemorten** [reply]
Great!

**Comment by paescuj** [comment]
The problem arises because you're deploying it to a different platform. When installing locally the package is fetched specifically for your local system (e.g. Windows / macOS), therefore you need to reinstall the dependencies on your server (which seems to be Linux) as opposed to simply copying the `node_modules` directory.

**Reply by thomasjonas** [reply]
I wasn't deploying to a different platform. I was just trying to setup Directus locally on my M1 macbook and followed the [Quickstart guide](https://docs.directus.io/self-hosted/quickstart.html) from the Self Hosted section.

**Reply by paescuj** [reply]
Okay, would you mind sharing the whole output from `npm init directus-project example-project`?

**Reply by thomasjonas** [reply]
Sure, no problem! I just ran `npm init directus-project example-project` and got the following output:

```
/Users/thomasjonas/Code/tmp/example-project/node_modules/snappy/index.js:1
"use strict";const{existsSync,readFileSync}=require("fs"),{join}=require("path"),{platform,arch}=process;let nativeBinding=null,localFileExisted=!1,loadError=null;function isMusl(){if(!process.report||typeof process.report.getReport!="function")try{return readFileSync("/usr/bin/ldd","utf8").includes("musl")}catch(e){return!0}else{const{glibcVersionRuntime:e}=process.report.getReport().header;return!e}}switch(platform){case"android":switch(arch){case"arm64":localFileExisted=existsSync(join(__dirname,"snappy.android-arm64.node"));try{localFileExisted?nativeBinding=require("./snappy.android-arm64.node"):nativeBinding=require("@napi-rs/snappy-android-arm64")}catch(e){loadError=e}break;case"arm":localFileExisted=existsSync(join(__dirname,"snappy.android-arm-eabi.node"));try{localFileExisted?nativeBinding=require("./snappy.android-arm-eabi.node"):nativeBinding=require("@napi-rs/snappy-android-arm-eabi")}catch(e){loadError=e}break;default:throw new Error(`Unsupported architecture on Android ${arch}`)}break;case"win32":switch(arch){case"x64":localFileExisted=existsSync(join(__dirname,"snappy.win32-x64-msvc.node"));try{localFileExisted?nativeBinding=require("./snappy.win32-x64-msvc.node"):nativeBinding=require("@napi-rs/snappy-win32-x64-msvc")}catch(e){loadError=e}break;case"ia32":localFileExisted=existsSync(join(__dirname,"snappy.win32-ia32-msvc.node"));try{localFileExisted?nativeBinding=require("./snappy.win32-ia32-msvc.node"):nativeBinding=require("@napi-rs/snappy-win32-ia32-msvc")}catch(e){loadError=e}break;case"arm64":localFileExisted=existsSync(join(__dirname,"snappy.win32-arm64-msvc.node"));try{localFileExisted?nativeBinding=require("./snappy.win32-arm64-msvc.node"):nativeBinding=require("@napi-rs/snappy-win32-arm64-msvc")}catch(e){loadError=e}break;default:throw new Error(`Unsupported architecture on Windows: ${arch}`)}break;case"darwin":switch(arch){case"x64":localFileExisted=existsSync(join(__dirname,"snappy.darwin-x64.node"));try{localFileExisted?nativeBinding=require("./snappy.darwin-x64.node"):nativeBinding=require("@napi-rs/snappy-darwin-x64")}catch(e){loadError=e}break;case"arm64":localFileExisted=existsSync(join(__dirname,"snappy.darwin-arm64.node"));try{localFileExisted?nativeBinding=require("./snappy.darwin-arm64.node"):nativeBinding=require("@napi-rs/snappy-darwin-arm64")}catch(e){loadError=e}break;default:throw new Error(`Unsupported architecture on macOS: ${arch}`)}break;case"freebsd":if(arch!=="x64")throw new Error(`Unsupported architecture on FreeBSD: ${arch}`);localFileExisted=existsSync(join(__dirname,"snappy.freebsd-x64.node"));try{localFileExisted?nativeBinding=require("./snappy.freebsd-x64.node"):nativeBinding=require("@napi-rs/snappy-freebsd-x64")}catch(e){loadError=e}break;case"linux":switch(arch){case"x64":if(isMusl()){localFileExisted=existsSync(join(__dirname,"snappy.linux-x64-musl.node"));try{localFileExisted?nativeBinding=require("./snappy.linux-x64-musl.node"):nativeBinding=require("@napi-rs/snappy-linux-x64-musl")}catch(e){loadError=e}}else{localFileExisted=existsSync(join(__dirname,"snappy.linux-x64-gnu.node"));try{localFileExisted?nativeBinding=require("./snappy.linux-x64-gnu.node"):nativeBinding=require("@napi-rs/snappy-linux-x64-gnu")}catch(e){loadError=e}}break;case"arm64":if(isMusl()){localFileExisted=existsSync(join(__dirname,"snappy.linux-arm64-musl.node"));try{localFileExisted?nativeBinding=require("./snappy.linux-arm64-musl.node"):nativeBinding=require("@napi-rs/snappy-linux-arm64-musl")}catch(e){loadError=e}}else{localFileExisted=existsSync(join(__dirname,"snappy.linux-arm64-gnu.node"));try{localFileExisted?nativeBinding=require("./snappy.linux-arm64-gnu.node"):nativeBinding=require("@napi-rs/snappy-linux-arm64-gnu")}catch(e){loadError=e}}break;case"arm":localFileExisted=existsSync(join(__dirname,"snappy.linux-arm-gnueabihf.node"));try{localFileExisted?nativeBinding=require("./snappy.linux-arm-gnueabihf.node"):nativeBinding=require("@napi-rs/snappy-linux-arm-gnueabihf")}catch(e){loadError=e}break;default:throw new Error(`Unsupported architecture on Linux: ${arch}`)}break;default:throw new Error(`Unsupported OS: ${platform}, architecture: ${arch}`)}if(!nativeBinding)throw loadError||new Error("Failed to load native binding");const{compressSync,compress,uncompressSync,uncompress}=nativeBinding;module.exports.compressSync=compressSync,module.exports.compress=compress,module.exports.uncompressSync=uncompressSync,module.exports.uncompress=uncompress;


Error: Cannot find module '@napi-rs/snappy-darwin-arm64'
Require stack:
- /Users/thomasjonas/Code/tmp/example-project/node_modules/snappy/index.js
- /Users/thomasjonas/Code/tmp/example-project/node_modules/directus/dist/utils/compress.js
- /Users/thomasjonas/Code/tmp/example-project/node_modules/directus/dist/cache.js
- /Users/thomasjonas/Code/tmp/example-project/node_modules/directus/dist/middleware/respond.js
- /Users/thomasjonas/Code/tmp/example-project/node_modules/directus/dist/controllers/activity.js
- /Users/thomasjonas/Code/tmp/example-project/node_modules/directus/dist/app.js
- /Users/thomasjonas/Code/tmp/example-project/node_modules/directus/dist/server.js
- /Users/thomasjonas/Code/tmp/example-project/node_modules/directus/dist/cli/index.js
- /Users/thomasjonas/Code/tmp/example-project/node_modules/directus/dist/cli/run.js
- /Users/thomasjonas/Code/tmp/example-project/node_modules/directus/cli.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:995:15)
    at Module._load (node:internal/modules/cjs/loader:841:27)
    at Module.require (node:internal/modules/cjs/loader:1061:19)
    at require (node:internal/modules/cjs/helpers:103:18)
    at Object.<anonymous> (/Users/thomasjonas/Code/tmp/example-project/node_modules/snappy/index.js:1:2284)
    at Module._compile (node:internal/modules/cjs/loader:1159:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1213:10)
    at Module.load (node:internal/modules/cjs/loader:1037:32)
    at Module._load (node:internal/modules/cjs/loader:878:12)
    at Module.require (node:internal/modules/cjs/loader:1061:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/Users/thomasjonas/Code/tmp/example-project/node_modules/snappy/index.js',
    '/Users/thomasjonas/Code/tmp/example-project/node_modules/directus/dist/utils/compress.js',
    '/Users/thomasjonas/Code/tmp/example-project/node_modules/directus/dist/cache.js',
    '/Users/thomasjonas/Code/tmp/example-project/node_modules/directus/dist/middleware/respond.js',
    '/Users/thomasjonas/Code/tmp/example-project/node_modules/directus/dist/controllers/activity.js',
    '/Users/thomasjonas/Code/tmp/example-project/node_modules/directus/dist/app.js',
    '/Users/thomasjonas/Code/tmp/example-project/node_modules/directus/dist/server.js',
    '/Users/thomasjonas/Code/tmp/example-project/node_modules/directus/dist/cli/index.js',
    '/Users/thomasjonas/Code/tmp/example-project/node_modules/directus/dist/cli/run.js',
    '/Users/thomasjonas/Code/tmp/example-project/node_modules/directus/cli.js'
  ]
}

Node.js v18.12.1
✖ Installing Directus
Error: undefined
npm ERR! code 1
npm ERR! path /Users/thomasjonas/Code/tmp
npm ERR! command failed
npm ERR! command sh -c -- create-directus-project example-project

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/thomasjonas/.npm/_logs/2022-11-23T17_09_33_627Z-debug-0.log
```

And the debug log:
```
0 verbose cli /Users/thomasjonas/.nvm/versions/node/v18.12.1/bin/node /Users/thomasjonas/.nvm/versions/node/v18.12.1/bin/npm
1 info using npm@8.19.2
2 info using node@v18.12.1
3 timing npm:load:whichnode Completed in 0ms
4 timing config:load:defaults Completed in 1ms
5 timing config:load:file:/Users/thomasjonas/.nvm/versions/node/v18.12.1/lib/node_modules/npm/npmrc Completed in 0ms
6 timing config:load:builtin Completed in 0ms
7 timing config:load:cli Completed in 1ms
8 timing config:load:env Completed in 0ms
9 timing config:load:file:/Users/thomasjonas/Code/tmp/.npmrc Completed in 0ms
10 timing config:load:project Completed in 1ms
11 timing config:load:file:/Users/thomasjonas/.npmrc Completed in 0ms
12 timing config:load:user Completed in 0ms
13 timing config:load:file:/Users/thomasjonas/.nvm/versions/node/v18.12.1/etc/npmrc Completed in 1ms
14 timing config:load:global Completed in 1ms
15 timing config:load:validate Completed in 0ms
16 timing config:load:credentials Completed in 0ms
17 timing config:load:setEnvs Completed in 0ms
18 timing config:load Completed in 4ms
19 timing npm:load:configload Completed in 4ms
20 timing npm:load:mkdirpcache Completed in 2ms
21 timing npm:load:mkdirplogs Completed in 0ms
22 verbose title npm init directus-project example-project
23 verbose argv "init" "directus-project" "example-project"
24 timing npm:load:setTitle Completed in 6ms
25 timing config:load:flatten Completed in 1ms
26 timing npm:load:display Completed in 2ms
27 verbose logfile logs-max:10 dir:/Users/thomasjonas/.npm/_logs
28 verbose logfile /Users/thomasjonas/.npm/_logs/2022-11-23T17_09_33_627Z-debug-0.log
29 timing npm:load:logFile Completed in 16ms
30 timing npm:load:timers Completed in 0ms
31 timing npm:load:configScope Completed in 0ms
32 timing npm:load Completed in 30ms
33 silly logfile start cleaning logs, removing 2 files
34 timing arborist:ctor Completed in 0ms
35 silly logfile done cleaning log files
36 http fetch GET 200 https://registry.npmjs.org/create-directus-project 363ms (cache revalidated)
37 timing arborist:ctor Completed in 0ms
38 timing arborist:ctor Completed in 0ms
39 timing command:init Completed in 53976ms
40 verbose stack Error: command failed
40 verbose stack     at ChildProcess.<anonymous> (/Users/thomasjonas/.nvm/versions/node/v18.12.1/lib/node_modules/npm/node_modules/@npmcli/promise-spawn/lib/index.js:63:27)
40 verbose stack     at ChildProcess.emit (node:events:513:28)
40 verbose stack     at maybeClose (node:internal/child_process:1091:16)
40 verbose stack     at ChildProcess._handle.onexit (node:internal/child_process:302:5)
41 verbose cwd /Users/thomasjonas/Code/tmp
42 verbose Darwin 21.6.0
43 verbose node v18.12.1
44 verbose npm  v8.19.2
45 error code 1
46 error path /Users/thomasjonas/Code/tmp
47 error command failed
48 error command sh -c -- create-directus-project example-project
49 verbose exit 1
50 timing npm Completed in 54055ms
51 verbose code 1
52 error A complete log of this run can be found in:
52 error     /Users/thomasjonas/.npm/_logs/2022-11-23T17_09_33_627Z-debug-0.log
```

**Reply by paescuj** [reply]
Thanks @thomasjonas! I think this might be related to `nvm`. Could you also share the output of `node -p process.arch`?

**Reply by thomasjonas** [reply]
`node -p process.arch` gives me `arm64` which is correct for my M1 Macbook.

**Reply by paescuj** [reply]
Hmm okay, thanks!

Somehow `npm` doesn't install '@napi-rs/snappy-darwin-arm64' while it is supposed to do that.
I guess the package is installed for a wrong platform. You could check that by running `ls node_modules/@napi-rs` inside `example-project`.
And I guess this is either related to `nvm` or this `npm` bug: https://github.com/npm/cli/issues/4828

Also stumbled upon this thread for similar issues with `nvm`: https://stackoverflow.com/questions/68896696/having-trouble-installing-npm-on-mac-m1
Although I'm not sure why one would need to change `arch`, it might still be worth to give it a try.
Maybe you can also try it with a newer version of `npm`.

**Reply by aiamk** [reply]
> Sure, no problem! I just ran `npm init directus-project example-project` and got the following output:
> 
> ```
> /Users/thomasjonas/Code/tmp/example-project/node_modules/snappy/index.js:1
> "use strict";const{existsSync,readFileSync}=require("fs"),{join}=require("path"),{platform,arch}=process;let nativeBinding=null,localFileExisted=!1,loadError=null;function isMusl(){if(!process.report||typeof process.report.getReport!="function")try{return readFileSync("/usr/bin/ldd","utf8").includes("musl")}catch(e){return!0}else{const{glibcVersionRuntime:e}=process.report.getReport().header;return!e}}switch(platform){case"android":switch(arch){case"arm64":localFileExisted=existsSync(join(__dirname,"snappy.android-arm64.node"));try{localFileExisted?nativeBinding=require("./snappy.android-arm64.node"):nativeBinding=require("@napi-rs/snappy-android-arm64")}catch(e){loadError=e}break;case"arm":localFileExisted=existsSync(join(__dirname,"snappy.android-arm-eabi.node"));try{localFileExisted?nativeBinding=require("./snappy.android-arm-eabi.node"):nativeBinding=require("@napi-rs/snappy-android-arm-eabi")}catch(e){loadError=e}break;default:throw new Error(`Unsupported architecture on Android ${arch}`)}break;case"win32":switch(arch){case"x64":localFileExisted=existsSync(join(__dirname,"snappy.win32-x64-msvc.node"));try{localFileExisted?nativeBinding=require("./snappy.win32-x64-msvc.node"):nativeBinding=require("@napi-rs/snappy-win32-x64-msvc")}catch(e){loadError=e}break;case"ia32":localFileExisted=existsSync(join(__dirname,"snappy.win32-ia32-msvc.node"));try{localFileExisted?nativeBinding=require("./snappy.win32-ia32-msvc.node"):nativeBinding=require("@napi-rs/snappy-win32-ia32-msvc")}catch(e){loadError=e}break;case"arm64":localFileExisted=existsSync(join(__dirname,"snappy.win32-arm64-msvc.node"));try{localFileExisted?nativeBinding=require("./snappy.win32-arm64-msvc.node"):nativeBinding=require("@napi-rs/snappy-win32-arm64-msvc")}catch(e){loadError=e}break;default:throw new Error(`Unsupported architecture on Windows: ${arch}`)}break;case"darwin":switch(arch){case"x64":localFileExisted=existsSync(join(__dirname,"snappy.darwin-x64.node"));try{localFileExisted?nativeBinding=require("./snappy.darwin-x64.node"):nativeBinding=require("@napi-rs/snappy-darwin-x64")}catch(e){loadError=e}break;case"arm64":localFileExisted=existsSync(join(__dirname,"snappy.darwin-arm64.node"));try{localFileExisted?nativeBinding=require("./snappy.darwin-arm64.node"):nativeBinding=require("@napi-rs/snappy-darwin-arm64")}catch(e){loadError=e}break;default:throw new Error(`Unsupported architecture on macOS: ${arch}`)}break;case"freebsd":if(arch!=="x64")throw new Error(`Unsupported architecture on FreeBSD: ${arch}`);localFileExisted=existsSync(join(__dirname,"snappy.freebsd-x64.node"));try{localFileExisted?nativeBinding=require("./snappy.freebsd-x64.node"):nativeBinding=require("@napi-rs/snappy-freebsd-x64")}catch(e){loadError=e}break;case"linux":switch(arch){case"x64":if(isMusl()){localFileExisted=existsSync(join(__dirname,"snappy.linux-x64-musl.node"));try{localFileExisted?nativeBinding=require("./snappy.linux-x64-musl.node"):nativeBinding=require("@napi-rs/snappy-linux-x64-musl")}catch(e){loadError=e}}else{localFileExisted=existsSync(join(__dirname,"snappy.linux-x64-gnu.node"));try{localFileExisted?nativeBinding=require("./snappy.linux-x64-gnu.node"):nativeBinding=require("@napi-rs/snappy-linux-x64-gnu")}catch(e){loadError=e}}break;case"arm64":if(isMusl()){localFileExisted=existsSync(join(__dirname,"snappy.linux-arm64-musl.node"));try{localFileExisted?nativeBinding=require("./snappy.linux-arm64-musl.node"):nativeBinding=require("@napi-rs/snappy-linux-arm64-musl")}catch(e){loadError=e}}else{localFileExisted=existsSync(join(__dirname,"snappy.linux-arm64-gnu.node"));try{localFileExisted?nativeBinding=require("./snappy.linux-arm64-gnu.node"):nativeBinding=require("@napi-rs/snappy-linux-arm64-gnu")}catch(e){loadError=e}}break;case"arm":localFileExisted=existsSync(join(__dirname,"snappy.linux-arm-gnueabihf.node"));try{localFileExisted?nativeBinding=require("./snappy.linux-arm-gnueabihf.node"):nativeBinding=require("@napi-rs/snappy-linux-arm-gnueabihf")}catch(e){loadError=e}break;default:throw new Error(`Unsupported architecture on Linux: ${arch}`)}break;default:throw new Error(`Unsupported OS: ${platform}, architecture: ${arch}`)}if(!nativeBinding)throw loadError||new Error("Failed to load native binding");const{compressSync,compress,uncompressSync,uncompress}=nativeBinding;module.exports.compressSync=compressSync,module.exports.compress=compress,module.exports.uncompressSync=uncompressSync,module.exports.uncompress=uncompress;
> 
> 
> Error: Cannot find module '@napi-rs/snappy-darwin-arm64'
> Require stack:
> - /Users/thomasjonas/Code/tmp/example-project/node_modules/snappy/index.js
> - /Users/thomasjonas/Code/tmp/example-project/node_modules/directus/dist/utils/compress.js
> - /Users/thomasjonas/Code/tmp/example-project/node_modules/directus/dist/cache.js
> - /Users/thomasjonas/Code/tmp/example-project/node_modules/directus/dist/middleware/respond.js
> - /Users/thomasjonas/Code/tmp/example-project/node_modules/directus/dist/controllers/activity.js
> - /Users/thomasjonas/Code/tmp/example-project/node_modules/directus/dist/app.js
> - /Users/thomasjonas/Code/tmp/example-project/node_modules/directus/dist/server.js
> - /Users/thomasjonas/Code/tmp/example-project/node_modules/directus/dist/cli/index.js
> - /Users/thomasjonas/Code/tmp/example-project/node_modules/directus/dist/cli/run.js
> - /Users/thomasjonas/Code/tmp/example-project/node_modules/directus/cli.js
>     at Module._resolveFilename (node:internal/modules/cjs/loader:995:15)
>     at Module._load (node:internal/modules/cjs/loader:841:27)
>     at Module.require (node:internal/modules/cjs/loader:1061:19)
>     at require (node:internal/modules/cjs/helpers:103:18)
>     at Object.<anonymous> (/Users/thomasjonas/Code/tmp/example-project/node_modules/snappy/index.js:1:2284)
>     at Module._compile (node:internal/modules/cjs/loader:1159:14)
>     at Module._extensions..js (node:internal/modules/cjs/loader:1213:10)
>     at Module.load (node:internal/modules/cjs/loader:1037:32)
>     at Module._load (node:internal/modules/cjs/loader:878:12)
>     at Module.require (node:internal/modules/cjs/loader:1061:19) {
>   code: 'MODULE_NOT_FOUND',
>   requireStack: [
>     '/Users/thomasjonas/Code/tmp/example-project/node_modules/snappy/index.js',
>     '/Users/thomasjonas/Code/tmp/example-project/node_modules/directus/dist/utils/compress.js',
>     '/Users/thomasjonas/Code/tmp/example-project/node_modules/directus/dist/cache.js',
>     '/Users/thomasjonas/Code/tmp/example-project/node_modules/directus/dist/middleware/respond.js',
>     '/Users/thomasjonas/Code/tmp/example-project/node_modules/directus/dist/controllers/activity.js',
>     '/Users/thomasjonas/Code/tmp/example-project/node_modules/directus/dist/app.js',
>     '/Users/thomasjonas/Code/tmp/example-project/node_modules/directus/dist/server.js',
>     '/Users/thomasjonas/Code/tmp/example-project/node_modules/directus/dist/cli/index.js',
>     '/Users/thomasjonas/Code/tmp/example-project/node_modules/directus/dist/cli/run.js',
>     '/Users/thomasjonas/Code/tmp/example-project/node_modules/directus/cli.js'
>   ]
> }
> 
> Node.js v18.12.1
> ✖ Installing Directus
> Error: undefined
> npm ERR! code 1
> npm ERR! path /Users/thomasjonas/Code/tmp
> npm ERR! command failed
> npm ERR! command sh -c -- create-directus-project example-project
> 
> npm ERR! A complete log of this run can be found in:
> npm ERR!     /Users/thomasjonas/.npm/_logs/2022-11-23T17_09_33_627Z-debug-0.log
> ```
> 
> And the debug log:
> 
> ```
> 0 verbose cli /Users/thomasjonas/.nvm/versions/node/v18.12.1/bin/node /Users/thomasjonas/.nvm/versions/node/v18.12.1/bin/npm
> 1 info using npm@8.19.2
> 2 info using node@v18.12.1
> 3 timing npm:load:whichnode Completed in 0ms
> 4 timing config:load:defaults Completed in 1ms
> 5 timing config:load:file:/Users/thomasjonas/.nvm/versions/node/v18.12.1/lib/node_modules/npm/npmrc Completed in 0ms
> 6 timing config:load:builtin Completed in 0ms
> 7 timing config:load:cli Completed in 1ms
> 8 timing config:load:env Completed in 0ms
> 9 timing config:load:file:/Users/thomasjonas/Code/tmp/.npmrc Completed in 0ms
> 10 timing config:load:project Completed in 1ms
> 11 timing config:load:file:/Users/thomasjonas/.npmrc Completed in 0ms
> 12 timing config:load:user Completed in 0ms
> 13 timing config:load:file:/Users/thomasjonas/.nvm/versions/node/v18.12.1/etc/npmrc Completed in 1ms
> 14 timing config:load:global Completed in 1ms
> 15 timing config:load:validate Completed in 0ms
> 16 timing config:load:credentials Completed in 0ms
> 17 timing config:load:setEnvs Completed in 0ms
> 18 timing config:load Completed in 4ms
> 19 timing npm:load:configload Completed in 4ms
> 20 timing npm:load:mkdirpcache Completed in 2ms
> 21 timing npm:load:mkdirplogs Completed in 0ms
> 22 verbose title npm init directus-project example-project
> 23 verbose argv "init" "directus-project" "example-project"
> 24 timing npm:load:setTitle Completed in 6ms
> 25 timing config:load:flatten Completed in 1ms
> 26 timing npm:load:display Completed in 2ms
> 27 verbose logfile logs-max:10 dir:/Users/thomasjonas/.npm/_logs
> 28 verbose logfile /Users/thomasjonas/.npm/_logs/2022-11-23T17_09_33_627Z-debug-0.log
> 29 timing npm:load:logFile Completed in 16ms
> 30 timing npm:load:timers Completed in 0ms
> 31 timing npm:load:configScope Completed in 0ms
> 32 timing npm:load Completed in 30ms
> 33 silly logfile start cleaning logs, removing 2 files
> 34 timing arborist:ctor Completed in 0ms
> 35 silly logfile done cleaning log files
> 36 http fetch GET 200 https://registry.npmjs.org/create-directus-project 363ms (cache revalidated)
> 37 timing arborist:ctor Completed in 0ms
> 38 timing arborist:ctor Completed in 0ms
> 39 timing command:init Completed in 53976ms
> 40 verbose stack Error: command failed
> 40 verbose stack     at ChildProcess.<anonymous> (/Users/thomasjonas/.nvm/versions/node/v18.12.1/lib/node_modules/npm/node_modules/@npmcli/promise-spawn/lib/index.js:63:27)
> 40 verbose stack     at ChildProcess.emit (node:events:513:28)
> 40 verbose stack     at maybeClose (node:internal/child_process:1091:16)
> 40 verbose stack     at ChildProcess._handle.onexit (node:internal/child_process:302:5)
> 41 verbose cwd /Users/thomasjonas/Code/tmp
> 42 verbose Darwin 21.6.0
> 43 verbose node v18.12.1
> 44 verbose npm  v8.19.2
> 45 error code 1
> 46 error path /Users/thomasjonas/Code/tmp
> 47 error command failed
> 48 error command sh -c -- create-directus-project example-project
> 49 verbose exit 1
> 50 timing npm Completed in 54055ms
> 51 verbose code 1
> 52 error A complete log of this run can be found in:
> 52 error     /Users/thomasjonas/.npm/_logs/2022-11-23T17_09_33_627Z-debug-0.log
> ```

i am facing the same exact issue on macos, i tried nodejs 16 & 18, same error.

**Comment by jorge-vnarvaez** [comment]
Any other workaround on this problem, i am facing the same problem i tried the solution answer but still doesn't work

**Reply by rijkvanzanten** [reply]
Use Docker instead of npm 👍🏻

**Reply by jorge-vnarvaez** [reply]
No chances to use npm?

**Reply by rijkvanzanten** [reply]
If you've already tried everything above to no success, I'd strongly recommend using Docker or Cloud instead 👍🏻

**Reply by jorge-vnarvaez** [reply]
Ok thank you very much I will considered to future projects, but already copy an existing project and change .env file and it works

**Comment by NecoHorne** [comment]
Anyone else that gets this, remove your `package-lock.json` from your repo and add it to `.gitignore` especially if your dev environment is different to your prod environment as `napi-rs` installs an os specific version when you run `npm install` and writes it to `package-lock.json`. Had the same issue (dev on windows and prod is running on ubuntu) you might also have to delete the napi-rs module from your node modules and re-run npm install but that should fix the issue.

**Comment by u12206050** [comment]
Have been running on GCP for months now without having this as a dependancy and suddenly today all builds are failing stating
`Error: Cannot find module '@napi-rs/snappy-linux-x64-gnu'` 

Have removed my package-lock.json to get it to work, but that didn't work either. Any one have any ideas?

**Reply by Voldemorten** [reply]
Have you tried installing the module manually? `npm i @napi-rs/snappy-linux-x64-gnu` in your directus project?

**Reply by u12206050** [reply]
Problem is I am on osx, so when I try that all I get is:
`Unsupported platform for @napi-rs/snappy-linux-x64-gnu@7.2.2: wanted {"os":"linux","arch":"x64"} (current: {"os":"darwin","arch":"arm64"})`

So not sure how I'm going to fix this. Until now I have just used my previous package-lock, but now I really need to update which changes it somehow...

**Reply by sambitraze** [reply]
try removing that and using this https://www.npmjs.com/package/@napi-rs/snappy-darwin-x64 instead

**Reply by u12206050** [reply]
The problem is I am deploying to GCP and it needs the linux version.
I have been able to install it now using `npm i @napi-rs/snappy-linux-x64-gnu -f` will see if it deploys

**Reply by u12206050** [reply]
It worked. :)
